File: 0875-koko-eating-bananas/0875-koko-eating-bananas.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        l, r = 1, max(piles)

        def total(k):
            ret = 0
            for i in piles:
                ret += math.ceil(i/k)
            return ret

        while l < r:
            m = (l + r) // 2
            if total(m) <= h: # 최소값 찾기니까 == 인 경우에도 더 최소인 값을 찾아보자
                r = m # 정답이 m일 수도 있기 때문에
            else:
                l = m + 1
        return l

File: 0875-koko-eating-bananas/0875-koko-eating-bananas.cpp
#include <cmath>
#include <algorithm>

class Solution {
public:
    int minEatingSpeed(vector<int>& piles, int h) {
        int l = 1, r = *max_element(piles.begin(), piles.end());

        while (l < r) {
            int m = (l + r) / 2;
            if (total(piles, m) <= h){
                r = m;
            } else {
                l = m + 1;
            }
        }

        return l;
    }

    int total(vector<int>& piles, int k) { // k가 커질 수록 total이 작아짐
        int ret = 0;
        for (auto i: piles) {
            ret += ceil((double)i/k);
        }
        return ret;
    }
};